lib/connection.js

```diff
@@ -27,607 +24,606 @@ let PACKET_HANDLERS = null;
+    }
 
-// Send a state synchronization packet over the connection
-//   path - path to a value in the application state
-//   verb - operation with this value (inc, dec, let, delete, push, pop, shift,
-//          unshift)
-//   value - a value to modify the current value with
-//
-Connection.prototype.notifyStateChange = function(path, verb, value) {
-  const packet = this.createPacket('state', path, verb, value);
-  this._send(packet);
-};
+    packet.callback[0] = packetId;
 
-// Send a handshake packet over the connection
-//   appName - name of an application to connect to
-//   login - user name (optional)
-//   password - user password (optional)
-//   callback - callback function to invoke after the handshake is completed
-//
-Connection.prototype.handshake = function(appName, login, password, callback) {
-  const packet = login && password ?
-    this.createPacket('handshake', appName, 'login', [login, password]) :
-    this.createPacket('handshake', appName);
-
-  const packetId = packet.handshake[0];
-  this._callbacks[packetId] = (error, sessionId) => {
-    if (login && password && !error) {
-      this.username = login;
-    }
-    this.sessionId = sessionId;
-    if (callback) {
-      callback(error, sessionId);
-    }
-  };
+    this._send(packet);
+  }
 
-  this._send(packet);
-};
+  // Send an event packet over the connection
+  //   interfaceName - name of an interface
+  //   eventName - name of an event
+  //   args - event arguments as an array
+  //
+  emitRemoteEvent(interfaceName, eventName, args) {
+    const packet = this.createPacket('event', interfaceName, eventName, args);
+    this._send(packet);
+  }
 
-// Send an inspect packet over the connection
-//   interfaceName - name of an interface to inspect
-//   callback - callback function to invoke after another side responds
-//              with interface introspection
-//
-Connection.prototype.inspectInterface = function(interfaceName, callback) {
-  const packet = this.createPacket('inspect', interfaceName, null, null);
-  const packetId = packet.inspect[0];
+  // Send a state synchronization packet over the connection
+  //   path - path to a value in the application state
+  //   verb - operation with this value (inc, dec, let, delete,
+  //          push, pop, shift, unshift)
+  //   value - a value to modify the current value with
+  //
+  notifyStateChange(path, verb, value) {
+    const packet = this.createPacket('state', path, verb, value);
+    this._send(packet);
+  }
 
-  this._callbacks[packetId] = (error, ...methods) => {
-    if (error) {
+  // Send a handshake packet over the connection
+  //   appName - name of an application to connect to
+  //   login - user name (optional)
+  //   password - user password (optional)
+  //   callback - callback function to invoke after the handshake is completed
+  //
+  handshake(appName, login, password, callback) {
+    const packet = login && password ? this.createPacket('handshake', appName,
```

Comment:
Can you please break the lines after `?` and `:` characters instead of between arguments?